torch/testing/_internal/distributed/_sharded_tensor/__init__.py

```diff
@@ -0,0 +1,85 @@
+import sys
+from functools import partial, wraps
+
+import torch
+import torch.distributed as dist
+from torch.distributed import rpc
+from torch.testing._internal.common_distributed import (
+    TEST_SKIPS,
+)
+
+
+class ShardedTensorTestBase(object):
+    @property
+    def world_size(self):
+        return 4
+
+    def init_pg(self):
+        dist.init_process_group(
+            backend="nccl",
+            world_size=self.world_size,
+            rank=self.rank,
+            init_method=f"file://{self.file_name}",
+        )
+
+    def init_rpc(self):
+        rpc_backend_options = rpc.TensorPipeRpcBackendOptions()
+        rpc_backend_options.init_method = f"file://{self.file_name}"
+        for rank in range(self.world_size):
+            rpc_backend_options.set_device_map(
+                f"worker{rank}", {rank: self.rank, self.rank: rank}
+            )
+
+        rpc.init_rpc(
+            name="worker%d" % self.rank,
+            rank=self.rank,
+            world_size=self.world_size,
+            rpc_backend_options=rpc_backend_options,
+        )
+
+    def init_comms(self, init_rpc=True):
+        if init_rpc:
+            self.init_rpc()
+        self.init_pg()
+
+    def destroy_comms(self, destroy_rpc=True):
+        # Wait for all ranks to reach here before starting shutdown.
+        dist.barrier()
+
+        if destroy_rpc:
+            rpc.shutdown()
+        dist.destroy_process_group()
+
+    def setUp(self) -> None:
+        super().setUp()
+        self._spawn_processes()
```

Comment:
where is _spawn_process defined?

Reply:
Ah good catch. This is actually defined in `MultiProcessTestCase` and this just worked accidentally since the tests inherited from ShardedTensorBase and also MultiProcessTestCase. I think ShardedTensorBase should inherit from MultiProcessTestCase.